docs/notebooks/ensembles-advanced.ipynb

```diff
@@ -0,0 +1,301 @@
+    "\n",
+    "The function`ensembles.plot_rsqprofile` provides plotting for evaluating the proportion of total variance in climate realizations that is covered by the selection.\n",
+    "\n",
+    "In this case ~78% of the total variance in original ensemble is covered by the selection."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ensembles.plot_rsqprofile(fig_data)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Alternatively we can use `method = {'rsq_cutoff':Float}` or `method = {'rsq_optimize':None}`\n",
+    "* For example with `rsq_cutoff` we instead find the number of realizations needed to cover the provided $R^{2}$ value"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ids1, cluster1, fig_data1 = ensembles.kmeans_reduce_ensemble(data=crit, method={'rsq_cutoff':0.85}, random_state=42, make_graph=True)\n",
+    "ensembles.plot_rsqprofile(fig_data1)\n",
+    "ds_crit.isel(realization=ids1)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### **KKZ reduce ensemble**\n",
+    "\n",
+    "`xclim` also makes available a similar ensemble reduction algorithm, `ensembles.kkz_reduce_ensemble`. The adavntage of this algorithm is largely that fewer realizations are needed in order to reach the same level of representative members than the K-means clustering algorithm, as the KKZ methods tends towards identifying members that fall towards the extremes of criteria values."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ids = ensembles.kkz_reduce_ensemble(crit, num_select=10)\n",
+    "ds_crit.isel(realization=ids)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plt.style.use('seaborn-dark')\n",
+    "fig = plt.figure(figsize=(9,9))\n",
+    "ax = plt.axes(projection='3d')\n",
+    "\n",
+    "for h in ds_crit.horizon:\n",
+    "\n",
+    "    ax.scatter(ds_crit.sel(horizon=h,realization=ids).delta_annual_tavg, ds_crit.sel(horizon=h, realization=ids).delta_annual_prtot ,\n",
+    "            ds_crit.sel(horizon=h,realization=ids).delta_JJA_prtot,label=f\"delta {h.values} - selected\")\n",
+    "\n",
+    "ax.set_xlabel('delta_annual_tavg (C)')\n",
+    "ax.set_ylabel('delta_annual_prtot (%)')\n",
+    "ax.set_zlabel('delta_JJA_prtot (%)')\n",
+    "plt.legend()\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To give a better sense of the differences between **Nested** and **Unnested** results, we can progressively identify members that would be chosen by each algorithm by iterative fashion."
```

Comment:
Add a sentence saying that KKZ provides nested results and KKZ not

Reply:
In reality I think the nested kkz way of doing things is such that users should typically increase the sample number by 2.   Where as k-means divides the space irregardless on n .

For example  KKZ always takes the median runs as its first choice, 2nd would be furthest from the median (say the biggest delta temp in the example) this would result in ignoring the 'cold' end of the delta spectrum altogether

On the other hand k-means clustering with n=2 would select 2 equally spaced runs (i.e kind of warm and kind of cool) with most likely neither being all that close  to the median value per se

This might be bit long winded, but I think it is an important point.  I am not sure the KKZ paper talks about this aspect but it is my personal 'expert' opinion 

edited for clarity

Reply:
@aulemahal does this make sense to you? you coded the algorithm but I am ~99% sure that i understand what is happening

Reply:
I,ve added (even) more graphs illustrating how low numbers can sometimes be tricky.... basically end up saying that user should be careful when selecting a small number of runs